db/dcrpg/pgblockchain.go

```diff
@@ -2365,69 +1984,49 @@ func (pgb *ChainDB) UpdateSpendingInfoInAllAddresses() (int64, error) {
 	}
 
 	updatesPerDBTx := 500
+	totalVinIbIDs := len(allVinDbIDs)
 
-	log.Infof("Updating spending tx info for %d addresses...", len(allVinDbIDs))
+	timeStart := time.Now()
+
+	log.Infof("Updating spending tx info for %d addresses...", totalVinIbIDs)
 	var numAddresses int64
-	for i := 0; i < len(allVinDbIDs); i += updatesPerDBTx {
-		//for i, vinDbID := range allVinDbIDs {
+	for i := 0; i < totalVinIbIDs; i += updatesPerDBTx {
 		if i%100000 == 0 {
 			endRange := i + 100000 - 1
-			if endRange > len(allVinDbIDs) {
-				endRange = len(allVinDbIDs)
+			if endRange > totalVinIbIDs {
+				endRange = totalVinIbIDs
 			}
 			log.Infof("Updating from vins %d to %d...", i, endRange)
 		}
 
-		/*var numAddressRowsSet int64
-		numAddressRowsSet, err = SetSpendingForVinDbID(pgb.db, vinDbID)
-		if err != nil {
-			log.Errorf("SetSpendingForFundingOP: %v", err)
-			continue
-		}
-		numAddresses += numAddressRowsSet*/
 		var numAddressRowsSet int64
 		endChunk := i + updatesPerDBTx
-		if endChunk > len(allVinDbIDs) {
-			endChunk = len(allVinDbIDs)
+		if endChunk > totalVinIbIDs {
+			endChunk = totalVinIbIDs
 		}
+
+		// fullMode is definately running so no need to check.
+		timeTakenPerBlock := (time.Since(timeStart).Seconds() / float64(endChunk-i))
+		timeToComplete := int64(timeTakenPerBlock * float64(totalVinIbIDs-endChunk))
```

Comment:
Because it not possible to accurately determine how the future will be (how long a given update will actually take) I have using the word `approx.` is telling the using there may be a variation in the time this update might take.

Reply:
Certainly it had to be an approximation, but the approximation can vary wildly.  When it hits a bunch of anomalous blocks, the estimate will jump up, then back down in the next more regular chunk of blocks.  It's the best way can do without smoothing out the estimate history, which would require a lot more code for little gain.

Reply:
Is there a way of doing this smoothing without resulting to a slower sync progress? Technically using the transactions estimation method was more accurate but would end up slowing down the sync.
This would be worse as it would give some more accuracy (not even close to 100%) and result to a slower sync. Consider this; syncing 500 blocks takes like 3 sec on average but the additional process of fetching the transactions count for a couple of last blocks sync'd and using that to estimate the average transaction counts per next bunch of 500 blocks will definitely take more than 5 sec to do a sync of 500 blocks on a slow machine. Take into consideration that this fetching of the transactions count for a couple of last blocks is going to happen at sync.go that doesn't have direct access to transactions count details thus slowing down the sync.
Because predicting the future close to accurate or even accurately here is almost impossible I would say the extra hustle of getting some more accurate values isn't worth it

Reply:
Not worth it right now anyway.